File: java/src/main/java/benchmarks/Decompression.java
package benchmarks;

import java.util.*;

public class Decompression extends Compression {
    
    private CompressedData compressedData;
    private byte[] decompressed;
    
    public Decompression() {
        sizeVal = configVal("size");
        resultVal = 0L;
    }
    
    @Override
    public String name() {
        return "Decompression";
    }
    
    @Override
    public void prepare() {
        testData = generateTestData(sizeVal);
        compressedData = compress(testData);
    }
    
    @Override
    public void run(int iterationId) {
        decompressed = decompress(compressedData);
        resultVal += decompressed.length;
    }
    
    @Override
    public long checksum() {
        long res = resultVal;
        if (Arrays.equals(testData, decompressed)) {
            res += 1000000;
        }
        return res;
    }
}

File: kotlin/src/main/kotlin/benchmarks/Decompression.kt
package benchmarks

import java.util.*

class Decompression : Compression() {
    private lateinit var compressedData: CompressedData
    private lateinit var decompressed: ByteArray
    
    init {
        // Уже инициализировано в родительском классе
    }
    
    override fun name(): String = "Decompression"
    
    override fun prepare() {
        testData = generateTestData(sizeVal)
        compressedData = compress(testData)
    }
    
    override fun run(iterationId: Int) {
        decompressed = decompress(compressedData)
        resultVal += decompressed.size.toUInt()
    }
    
    override fun checksum(): UInt {
        var res = resultVal
        if (testData.contentEquals(decompressed)) {
            res += 1000000u
        }
        return res
    }
}